phoenix-core/src/it/java/org/apache/phoenix/end2end/ViewUtilIT.java

```diff
@@ -333,4 +356,120 @@ public void testFindAllRelativesForTenantConnection() throws Exception {
+                        + "CONSTRAINT PK PRIMARY KEY (TENANT_ID, KP)) MULTI_TENANT=true";
+
+        String
+                globalViewDDL =
+                "CREATE VIEW IF NOT EXISTS " + globalViewName + "(G1 BIGINT, G2 BIGINT) "
+                        + "AS SELECT * FROM " + multiTenantTableName + " WHERE KP = '001' VIEW_TTL=30";
+
+        String viewDDL = "CREATE VIEW IF NOT EXISTS %s (V1 BIGINT, V2 BIGINT) AS SELECT * FROM %s VIEW_TTL=30";
+        String
+                viewIndexDDL =
+                "CREATE INDEX IF NOT EXISTS " + tenantViewIndex + " ON " + tenantViewOnGlobalView
+                        + "(NUM DESC) INCLUDE (ID)";
+
+        String
+                customObjectViewDDL =
+                "CREATE VIEW IF NOT EXISTS %s (V1 BIGINT, V2 BIGINT) " + "AS SELECT * FROM "
+                        + multiTenantTableName + " WHERE KP = '%s' VIEW_TTL=30";
+
+        String selectFromViewSQL = "SELECT * FROM %s";
+
+        List<String> dmls = Arrays.asList(new String[] {
+                String.format(viewDDL, tenantViewOnGlobalView, globalViewName),
+                String.format(customObjectViewDDL, customObjectView1, view1),
+                String.format(customObjectViewDDL, customObjectView2, view2),
+                viewIndexDDL
+        });
+
+        // Create the various tables and views
+        try (Connection conn = DriverManager.getConnection(getUrl())) {
+            // Base table.
+            try (Statement stmt = conn.createStatement()) {
+                stmt.execute(multiTenantTableDDL);
+            }
+            // Global view.
+            try (Statement stmt = conn.createStatement()) {
+                stmt.execute(globalViewDDL);
+            }
+            // Tenant views and indexes.
+            try (Connection tenantConn = DriverManager.getConnection(getUrl(), tenantProps)) {
+                for (String dml : dmls) {
+                    try (Statement stmt = tenantConn.createStatement()) {
+                        stmt.execute(dml);
+                    }
+                }
+            }
+        }
+
+        List<String> sqls = Arrays.asList(new String[] {
+                String.format(selectFromViewSQL, tenantViewOnGlobalView),
+                String.format(selectFromViewSQL, customObjectView1),
+                String.format(selectFromViewSQL, customObjectView2),
+                String.format(selectFromViewSQL, tenantIndex)
+        });
+
+        // Validate the view where clause for the above sqls.
+        try (Connection tenantConn = DriverManager.getConnection(getUrl(), tenantProps)) {
+            for (String sql : sqls) {
+                try (Statement stmt = tenantConn.createStatement()) {
+                    PhoenixStatement pstmt = stmt.unwrap(PhoenixStatement.class);
+                    QueryPlan plan = pstmt.compileQuery(sql);
+                    validatePhoenixRowTimestampParseNodeExists(plan);
+                }
+            }
+        }
+
+    }
+
```

Comment:
can we add one more test for view without ttl but calling `getViewWhereWithPhoenixTTL` method to validate the right behavior?